manager/dispatcher/dispatcher.go

```diff
@@ -316,6 +317,7 @@ func (d *Dispatcher) markNodesUnknown(ctx context.Context) error {
 				node.Status = api.NodeStatus{
 					State:   api.NodeStatus_UNKNOWN,
 					Message: `Node moved to "unknown" state due to leadership change in cluster`,
+					Address: node.Status.Address,
```

Comment:
This is fine but I'd suggest changing the code to:

```go
node.Status.State = api.NodeStatus_UNKNOWN
node.Status.Message = `Node moved to "unknown" state due to leadership change in cluster`
```

That way, if we add additional fields to `NodeStatus`, they won't be overwritten here.